server/models/Backlink.ts

```diff
@@ -37,6 +55,67 @@ class Backlink extends IdModel<
   @ForeignKey(() => Document)
   @Column(DataType.UUID)
   reverseDocumentId: string;
+
+  public static async findSourceDocumentIdsForUser(
```

Comment:
This works but is quite complicated, I'm worried about bugs – given that `Document.findByPk` already has all the logic to scope to a user in just these lines:

```
https://github.com/outline/outline/blob/7a8d40b9e73454ee70cc5f4320d9e49ba0f48f3c/server/models/Document.ts#L687-L698
```

could we create something similar, perhaps `Document.findByIds` and just pass in the backlink document ID's?

Reply:
Agreed - We can filter for membership existence for the results from `Document.findByIds` (since the query uses left outer join).

Reply:
looking much better